artifactory/services/utils/artifactoryutils.go

```diff
@@ -178,6 +178,42 @@ func parseBuildNameAndNumber(buildIdentifier string) (buildName string, buildNum
 	return buildName, buildNumber
 }
 
+func getBundleNameAndNumberFromBundleIdentifier(bundleIdentifier string) (string, string, error) {
```

Comment:
I suggest renaming this function to "parseBundleNameAndVersion", to be compatible with the exiting "parseBuildNameAndNumber" function.
The two function also have some code can be reused. For example. the following loop, but I think there's more:
```
	for i := len(bundle) - 1; i >= 1; i-- {
		bundleVersionsArray = append([]string{bundle[i]}, bundleVersionsArray...)
		if !strings.HasSuffix(bundle[i-1], EscapeChar) {
			bundleName = strings.Join(bundle[:i], Delimiter)
			bundleVersion = strings.Join(bundleVersionsArray, Delimiter)
			break
		}
	}
```